### PR TITLE
Select background for scene

### PR DESCRIPTION
**Pull request by CFarcy**

### CODE REVIEW

File: terra_layer/serializers.py [hunk 1]
```diff
@@ -39,6 +48,14 @@ def get_icon(self, obj):
     class Meta:
         model = Scene
         fields = "__all__"
+        extra_kwargs = {"baselayer": {"allow_empty": True}}
+
+    def to_internal_value(self, data):
+        baselayer = data.get("baselayer")
+        querydict = data.copy()
```
Comment (jrmi) [inline terra_layer/serializers.py:55]: You should copy only if type base layer is str, no ?

Comment (CFarcy) [reply terra_layer/serializers.py:55]: Yes, you're right. I did it that way to have only one "return variable". I will change it

Comment (CFarcy) [reply terra_layer/serializers.py:59]: Done.

File: terra_layer/urls/__init__.py [hunk 2]
```diff
@@ -17,6 +18,7 @@
     # Extra urls from third part modules
     path("", include("terra_accounts.urls")),
     path("", include("terra_utils.urls")),
+    path("", include("mapbox_baselayer.urls")),
```
Comment (jrmi) [inline terra_layer/urls/__init__.py:21]: What do you think of letting the module baselayer include his urls by himself ?

Comment (jrmi) [reply terra_layer/urls/__init__.py:21]: what do you think @submarcos ?

File: terra_layer/serializers.py [hunk 3]
```diff
@@ -7,6 +10,12 @@
 from .models import CustomStyle, FilterField, Layer, Scene
 
 
+class BaseLayerSerializer(ModelSerializer):
```
Comment (jrmi) [inline terra_layer/serializers.py:13]: Should be in baselayer module ?

File: terra_layer/views/layers.py [hunk 4]
```diff
@@ -477,3 +501,9 @@ def layers(self):
         if layers:
             return layers
         raise Http404
+
+
+class BaseLayerViewSet(ModelViewSet):
```
Comment (jrmi) [inline terra_layer/views/layers.py:506]: Should be in baselayer module too ?

File: terra_layer/urls/__init__.py (hunk 2 above)
Comment (submarcos) [reply terra_layer/urls/__init__.py:21]: It depends on project policy. Its is just a non mandatory helper to provide an unauthenticated way to serve Base layer tilejson. If project require more protection, it is easy to serve these urls in an authenticated view. But I think mapbox and map js frameworks dont like JWT

File: terra_layer/serializers.py (hunk 3 above)
Comment (submarcos) [reply terra_layer/serializers.py:13]: Base layer module is not a terra module, and can be used in every project using mapbox in frontend. I dont think adding requirements to DRF in this module is a good idea. It is not required to write a serializer  for this, just using a reverse url for tilejson in another serializer (scene ?) should be sufficient

Comment (submarcos) [reply terra_layer/serializers.py:13]: https://github.com/Terralego/django-terra-geocrud/blob/master/terra_geocrud/serializers.py#L90

Comment (submarcos) [reply terra_layer/serializers.py:13]: https://github.com/Terralego/django-terra-geocrud/blob/master/terra_geocrud/views.py#L88

File: terra_layer/views/layers.py (hunk 4 above)
Comment (CFarcy) [reply terra_layer/views/layers.py]: It has been moved in another django app set as a dependency for this one.

File: terra_layer/serializers.py (hunk 3 above)
Comment (CFarcy) [reply terra_layer/serializers.py]: It has been moved in another django app set as a dependency for this one (django-terra-settings).

File: terra_layer/urls/__init__.py (hunk 2 above)
Comment (jrmi) [reply terra_layer/urls/__init__.py:20]: For the moment, we can keep it here. We'll see if we need to change.

**Review by jrmi — APPROVED** [review]

**Pull request merged**